core/src/term/mod.rs

```diff
@@ -884,7 +919,11 @@ impl Term {
             Term::Bool(_) => Some("Bool".to_owned()),
             Term::Num(_) => Some("Number".to_owned()),
             Term::Str(_) => Some("String".to_owned()),
-            Term::Fun(_, _) | Term::FunPattern(_, _) => Some("Function".to_owned()),
+            Term::Fun(_, _)
+            | Term::FunPattern(_, _)
+            // We could print a separate type for predicates. For the time being, we just consider
+            // it to be the function resulting of `$predicate_to_ctr pred`.
+            | Term::CustomContract(CustomContract::Predicate(..)) => Some("Function".to_owned()),
```

Comment:
Since the new node is supposed to enable more precise errors, why not "Contract" here?

Reply:
Fixed in a follow-up coming PR.